[Multicolumn] Handle multicol intrinsic inline-size changes

Ahmad-S792 opened the pull request:
#### 340b0f1155192cc2bddf1df560b345a56a4b2890
<pre>
[Multicolumn] Handle multicol intrinsic inline-size changes

<a href="https://bugs.webkit.org/show_bug.cgi?id=257027">https://bugs.webkit.org/show_bug.cgi?id=257027</a>

Reviewed by Alan Baradlay.

This patch aligns WebKit with Blink / Chromium and Gecko / Firefox.

Merge: <a href="https://chromium.googlesource.com/chromium/src.git/+/209ad9840b0152a427e9d2b29318cfa47ca80d92">https://chromium.googlesource.com/chromium/src.git/+/209ad9840b0152a427e9d2b29318cfa47ca80d92</a>

Currently we used to store intrinsic size in column sets but if any
change happen, &apos;column sets&apos; do not get dirtied so this patch will
reset intrinsic size of column sets to 0 to fix this.

* Source/WebCore/rendering/RenderFragmentContainer.cpp:
(RenderFragmentContainer::computeIntrinsicLogicalWidths): As above
* LayoutTests/TestExpectations: Remove &apos;failing&apos; test

Canonical link: <a href="https://commits.webkit.org/264292@main">https://commits.webkit.org/264292@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d834fe1aa88d7407defa75be821f2f761ae9718

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7325 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7581 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7755 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8951 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7530 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7335 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9020 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7507 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/10423 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7454 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/8225 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6742 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9060 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5559 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6665 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14389 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/7106 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6771 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9781 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7260 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/5924 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6614 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10818 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/849 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6998 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderFragmentContainer.cpp:
```diff
@@ -373,9 +374,7 @@ void RenderFragmentContainer::computeIntrinsicLogicalWidths(LayoutUnit& minLogic
-
-    minLogicalWidth = m_fragmentedFlow->minPreferredLogicalWidth();
-    maxLogicalWidth = m_fragmentedFlow->maxPreferredLogicalWidth();
+    maxLogicalWidth = minLogicalWidth = LayoutUnit();
```
WebKit's style is as follows:
maxLogicalWidth = { }; 
minLogicalWidth = { };

Ahmad-S792 replied:
Cool! Will push updated commit once back (out right now). 👍

Ahmad-S792 replied:
Done.

Ahmad-S792 committed:
[Multicolumn] Handle multicol intrinsic inline-size changes

https://bugs.webkit.org/show_bug.cgi?id=257027

Reviewed by Alan Baradlay.

This patch aligns WebKit with Blink / Chromium and Gecko / Firefox.

Merge: https://chromium.googlesource.com/chromium/src.git/+/209ad9840b0152a427e9d2b29318cfa47ca80d92

Currently we used to store intrinsic size in column sets but if any
change happen, 'column sets' do not get dirtied so this patch will
reset intrinsic size of column sets to 0 to fix this.

* Source/WebCore/rendering/RenderFragmentContainer.cpp:
(RenderFragmentContainer::computeIntrinsicLogicalWidths): As above
* LayoutTests/TestExpectations: Remove 'failing' test

Canonical link: https://commits.webkit.org/264292@main
